src/web_app/js/call.js

```diff
@@ -419,23 +419,45 @@ Call.prototype.maybeCreatePcClient_ = function() {
   if (this.pcClient_) {
     return;
   }
-  try {
-    this.pcClient_ = new PeerConnectionClient(this.params_, this.startTime);
-    this.pcClient_.onsignalingmessage = this.sendSignalingMessage_.bind(this);
-    this.pcClient_.onremotehangup = this.onremotehangup;
-    this.pcClient_.onremotesdpset = this.onremotesdpset;
-    this.pcClient_.onremotestreamadded = this.onremotestreamadded;
-    this.pcClient_.onsignalingstatechange = this.onsignalingstatechange;
-    this.pcClient_.oniceconnectionstatechange = this.oniceconnectionstatechange;
-    this.pcClient_.onnewicecandidate = this.onnewicecandidate;
-    this.pcClient_.onerror = this.onerror;
-    trace('Created PeerConnectionClient');
-  } catch (e) {
-    this.onError_('Create PeerConnection exception: ' + e.message);
-    alert('Cannot create RTCPeerConnection; ' +
-        'WebRTC is not supported by this browser.');
-    return;
-  }
+  return new Promise(function(resolve, reject) {
+    if (typeof RTCPeerConnection.generateCertificate === 'function') {
+      var certParams = {name: 'ECDSA', namedCurve: 'P-256'};
+      RTCPeerConnection.generateCertificate(certParams)
+      .then(function(cert) {
+        trace('ECDSA Certificate generated successfully.');
+        this.params_.peerConnectionConfig.certificates = [cert];
+        this.createPcClient_();
+        resolve();
+      }.bind(this))
+      .catch(function(error) {
+        if (this.params_.peerConnectionConfig.certificates) {
+          reject(error);
+        } else {
+          // This is not a critical error hence why it should continue
```

Comment:
I did this because I did not want a certificate generation failure (we would fallback to the default certificate right?) to prevent the call from being setup but still catch other errors which most likely are fatal.